Amazon.QLDB.Driver.IntegrationTests/AsyncStatementExecutionTests.cs

```diff
@@ -0,0 +1,144 @@
+
+            integrationTestBase.RunForceDeleteLedger();
+
+            integrationTestBase.RunCreateLedger();
+            qldbDriver = integrationTestBase.CreateDriver(amazonQldbSessionConfig);
+
+            // Create table.
+            var query = $"CREATE TABLE {Constants.TableName}";
+
+            var count = await qldbDriver.Execute(async txn =>
+            {
+                var result = await txn.Execute(query);
+
+                var count = 0;
+                await foreach (var row in result)
+                {
+                    count++;
+                }
+                return count;
+            });
+
+
+            Assert.AreEqual(1, count);
+
+            var result = await qldbDriver.ListTableNamesAsync();
+            foreach (var row in result)
+            {
+                Assert.AreEqual(Constants.TableName, row);
+            }
+        }
+
+        [ClassCleanup]
+        public static void ClassCleanup()
+        {
+            integrationTestBase.RunDeleteLedger();
+            qldbDriver.Dispose();
+        }
+
+        [TestCleanup]
+        public async void TestCleanup()
+        {
+            // Delete all documents in table.
+            await qldbDriver.Execute(txn => txn.Execute($"DELETE FROM {Constants.TableName}"));
+        }
+
+        [TestMethod]
+        public async void Execute_DropExistingTable_TableDropped()
+        {
+            // Given.
+            var create_table_query = $"CREATE TABLE {Constants.CreateTableName}";
+            var create_table_count = await qldbDriver.Execute(async txn =>
+            {
+                var result = await txn.Execute(create_table_query);
+
+                var count = 0;
+                await foreach (var row in result)
+                {
+                    count++;
+                }
+                return count;
+            });
+            Assert.AreEqual(1, create_table_count);
+
+            // Execute ListTableNames() to ensure table is created.
+            var result = await qldbDriver.ListTableNamesAsync();
+
+            var tables = new List<string>();
+            foreach (var row in result)
+            {
+                tables.Add(row);
+            }
+            Assert.IsTrue(tables.Contains(Constants.CreateTableName));
+
+            // When.
+            var drop_table_query = $"DROP TABLE {Constants.CreateTableName}";
+            var drop_table_count = await qldbDriver.Execute(async txn =>
+            {
+                var result = await txn.Execute(drop_table_query);
+
+                var count = 0;
+                await foreach (var row in result)
+                {
+                    count++;
+                }
+                return count;
+            });
+            Assert.AreEqual(1, drop_table_count);
+
+            // Then.
+            tables.Clear();
+            var updated_tables_result = await qldbDriver.ListTableNamesAsync();
+            foreach (var row in updated_tables_result)
```

Comment:
await

Reply:
Same here since 'updated_tables_result' is an IEnumerable<string>, we cannot await for it in the foreach statement unless it's an IAsyncEnumerable<string>.